### PR TITLE
Laurent 1973

### PR DESCRIPTION
**Pull request by LaurentPV**
Adding the "Remember Profile" feature as described in Issue #1973

### CODE REVIEW

File: shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java [hunk 1]
```diff
@@ -31,6 +32,26 @@ public void actionPerformed(ActionEvent e) {
...
+			logger.info("Saving Profile selected in basic.properties file.");
+			String fileName = ShUpConfig.shanoirUploaderFolder + File.separator + ShUpConfig.BASIC_PROPERTIES;
+			final File propertiesFile = new File(fileName);
+			boolean propertiesExists = propertiesFile.exists();
+			if (propertiesExists) {
+				try {
+					Properties props = ShUpConfig.basicProperties;
+					props.setProperty(ShUpConfig.PROFILE, selectedProfile);
+					// Store the new Profile configuration in the basic.properties file
+					OutputStream out = new FileOutputStream(propertiesFile);
```
Comment (michaelkain) [inline shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java:48]: Hi Laurent, good PR, after my code review. I will have to test locally, but it looks very promising.
Please put try (new FileOutputStream) to close the stream correctly? Thank you and with kind regards, Michael

Comment (LaurentPV) [reply shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java:48]: I just pushed a commit with an out.close(); at the end of the try.

Comment (michaelkain) [reply shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java:48]: Hi Laurent, thank you! I would propose this: try (OutputStream out = new FileOutputStream(propertiesFile)) {
xxx }, to be sure, as with finally before, that the stream is closed at any moment. Hope it works!

Comment (michaelkain) [reply shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java:48]: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (LaurentPV) [reply shanoir-uploader/src/main/java/org/shanoir/uploader/action/init/SelectProfilePanelActionListener.java]: Thanks, I just pushed the correction

**Review by michaelkain — COMMENTED** [review]
Great, thx, code review finished. I will have to test locally.

**Review by michaelkain — APPROVED** [review]
Perfect, my local tests work! I tested with two different profiles. Nice job!

**Pull request merged**